wgpu-hal/src/dx12/device.rs

```diff
@@ -349,6 +354,7 @@ impl crate::Device<super::Api> for super::Device {
         Ok(super::Buffer {
             resource,
             size,
+            initialized: AtomicBool::new(allocation.is_none()),
```

Comment:
Under the belief this is a bug in the Xe drivers, we should only be calling discard resource on the Xe drivers, and we should be doing it unconditionally for transiant buffers on their creation.

Reply:
I prefer this to be the difference between `PlacedResource` and `CommittedResource`, because even on Xe, there is no problem using `CommittedResource`.

Reply:
Conditional on both xe and suballocation is fine